plugins/references.py

```diff
@@ -541,9 +541,21 @@ def run(self, edit):
                     flatrefs.append((name, view.substr(line_reg).strip("\n")))
                 sel.add(line_reg)
 
+        sorting_funcs = {
+            "marker_order": lambda x: marker_order[x[0].lower()]
+            if x[0].lower() in marker_order
+            else 9999,
+            "alphabetical": lambda x: x[0].lower(),
+            "numeric": lambda x: [
+                int(p) if p.isnumeric() else p for p in re.split(r"[ _.-]", x[0].lower())
+            ],
+        }
+        settings = view.settings()
```

Comment:
If `alphabetical` or `numeric` is selected, `marker_order` object is comuted for nothing. I'd prefer a solution which rallc line 525 to 528 only if required.

Reply:
You could definitely optimize this slightly with something like this, but possibly at the cost of cleanness:

```python
        ...
        marker_order = None
        
        def makerOrderSort(x):
            nonlocal marker_order
            if marker_order is None:
                markers = getMarkers(view)
                marker_order = sorted(
                    markers.keys(), key=lambda marker: min(markers[marker].regions, key=lambda reg: reg.a).a
                )
                marker_order = dict(zip(marker_order, range(0, len(marker_order))))
            return marker_order[x[0].lower()] if x[0].lower() in marker_order else 9999

        sorting_funcs = {
            "marker_order": lambda x: makerOrderSort,
            "alphabetical": lambda x: x[0].lower(),
            "numeric": lambda x: [
                int(p) if p.isnumeric() else p for p in re.split(r"[ _.-]", x[0].lower())
            ],
        }
        ...
```

Reply:
I guess we can keep it as is because `marker_order` turned out to be used to print missing links as well.